src/invision/components/movie-list/movie-list.js

```diff
@@ -1,90 +1,167 @@
-import React from 'react';
-import thumbnail from './thumbnail.png';
+import React, { useState, useCallback, useMemo } from 'react';
+
+import MoviePopup from '../movie-popup/movie.popup';
+import DeletePopup from '../delete-movie/delete-popup';
+
 import MovieItem from '../movie-item/movie-item'
+import SearchBar from '../search-bar/search-bar';
+import MovieDetail from '../movie-detail/movie-detail';
 import './movie-list.scss';
+
 import MovieService from '../../services/movie.service';
+import useFetchMovie from './fetch-movie.hook';
 
-export default class MovieList extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      movieList: [],
-      sortValue: 'RELEASE DATE',
-    };
 
-    this.getMovieList();
+export default function MovieList() {
 
-    this.sort = this.sort.bind(this);
-  }
+  const [movieList, updateMovieList] = useState([]);
+  const [fullMovieList, updateFullMovieList] = useState([]);
+  const [sortValue, updateSortValue] = useState('RELEASE DATE');
+  const [isEditMovieVisible, updateMoviePopupVisibility] = useState(false);
+  const [isDeleteMovieVisible, updateDeleteMoviePopupVisibility] = useState(false);
+  const [selectedMovie, updateSelectedMovie] = useState(null);
+  const [detailedMovie, updateDetailedMovie] = useState(null);
 
-  getMovieList() {
-    new MovieService().getMovieList().then(movieList => {
-      this.setState({ movieList });
+  useFetchMovie(() => {
+    new MovieService().getMovieList().then(newMovieList => {
+      updateMovieList(newMovieList);
+      updateFullMovieList(newMovieList);
     });
+  });
+
+  const onHandleSearchClick = (searchKeyword) => {
+    const searchedMovies = fullMovieList.filter(movie => movie.title.toLowerCase().includes(searchKeyword.toLowerCase()));
+    updateMovieList(searchedMovies);
   }
 
-  sort(ev) {
-    this.setState({ sortValue: ev.target.value });
+  const filterCategory = useCallback((categoryName = '') => {
```

Comment:
useCallback usually used to prevent rerender od the child components. It works because you put the same function to the child component. In case of filterCategory you do not get any benefit of using useCallback hook.